packages/mask/src/extension/popups/pages/Wallet/EditNetwork/index.tsx

```diff
@@ -110,79 +109,122 @@ export const EditNetwork = memo(function EditNetwork() {
             </Button>,
         )
         return () => setExtension(undefined)
-    }, [isBuiltIn, chainId, classes.iconButton])
+    }, [isBuiltIn, id, classes.iconButton])
 
     const schema = useMemo(() => createSchema(t), [t])
 
     type FormInputs = z.infer<typeof schema>
     const {
-        handleSubmit,
+        getValues,
         register,
         setError,
-        formState: { errors, isValidating, isSubmitting, isValid: isFormValid },
+        formState: { errors, isValidating, isDirty },
     } = useForm<FormInputs>({
         mode: 'all',
         resolver: zodResolver(schema),
         defaultValues: network,
     })
     const { showSnackbar } = usePopupCustomSnackbar()
+    const checkZodError = useCallback(
+        (message: string) => {
+            try {
+                const issues = JSON.parse(message) as ZodCustomIssue[]
+                const isInvalid = issues.some((issue) => issue.path[0] !== 'currencySymbol')
+                if (!isInvalid) return true
+                issues.forEach((issue) => {
+                    // We assume there is no multiple paths.
+                    setError(issue.path[0] as keyof FormInputs, {
+                        message: t(issue.message as AvailableLocaleKeys, issue.params),
+                    })
+                })
+            } catch {}
```

Comment:
`setError` in the catch clauses.

Reply:
`setError` requires a field name.
This try-catch is to parse the json safely, it it fails we can ignore it.